frontend/packages/integration-tests-cypress/tests/app/console-plugin-manifest.cy.ts

```diff
@@ -0,0 +1,273 @@
+                .should('contain', 'created'),
+            )
+            .then(() => {
+              cy.visit(`/k8s/ns/${PLUGIN_NAME}/deployments`);
+              listPage.rows.shouldBeLoaded();
+              listPage.filter.byName(PLUGIN_NAME);
+              listPage.rows.shouldExist(PLUGIN_NAME);
+              enableDemoPlugin(true);
+            });
+        } else {
+          console.log('this IS A local env, not setting the pull spec for the deployment');
+        }
+      });
+    });
+
+    beforeEach(() => {
+      // Visit the console plugins list page to ensure plugin exists and is loaded
+      cy.visit('/k8s/cluster/console.openshift.io~v1~ConsolePlugin');
+      cy.get('body').then(($body) => {
+        if ($body.find(`[data-test-id="${PLUGIN_NAME}"]`).length > 0) {
+          // Plugin exists, proceed with tests
+          cy.byLegacyTestID(PLUGIN_NAME).should('be.visible');
+        } else {
+          // Skip tests if plugin doesn't exist
+          cy.log(`Plugin ${PLUGIN_NAME} not found, skipping manifest tests`);
+        }
+      });
+    });
+
+    afterEach(() => {
+      checkErrors();
+    });
+
+    after(() => {
+      if (!isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+        enableDemoPlugin(false);
+      }
+      cy.deleteProjectWithCLI(PLUGIN_NAME);
+    });
+
+    it('should display manifest tab in ConsolePlugin details page', () => {
+      // Navigate to the demo plugin details page
+      cy.visit(`/k8s/cluster/console.openshift.io~v1~ConsolePlugin/${PLUGIN_NAME}`);
+
+      // Verify we're on the plugin details page
+      detailsPage.titleShouldContain(PLUGIN_NAME);
+
+      // Check that the Plugin manifest tab exists
+      cy.get('[role="tablist"]').within(() => {
+        cy.contains('Plugin manifest').should('be.visible');
+      });
+    });
+
+    it('should navigate to manifest tab and display content', () => {
+      // Navigate directly to the manifest tab
+      cy.visit(`/k8s/cluster/console.openshift.io~v1~ConsolePlugin/${PLUGIN_NAME}/plugin-manifest`);
+
+      // Verify we're on the manifest tab
+      cy.url().should('include', '/plugin-manifest');
+
+      // Verify the manifest tab is active (PatternFly v6 uses pf-m-current class)
+      cy.get('[role="tablist"]').within(() => {
+        cy.contains('Plugin manifest').parent().should('have.class', 'pf-m-current');
+      });
+
+      // Wait for the page to load
+      detailsPage.isLoaded();
+
+      // Check if manifest content is displayed
+      cy.get('body').then(($body) => {
+        if ($body.find('.monaco-editor').length > 0) {
+          // Code editor is present - verify it contains JSON content
+          cy.get('.monaco-editor').should('be.visible');
+
+          // Verify the editor contains typical plugin manifest structure
+          cy.get('.monaco-editor').within(() => {
+            // Look for common manifest properties
+            cy.get('.view-lines').should('contain.text', '"name"');
+          });
```

Comment:
CodeEditor exposes `window.monaco` for more comprehensive testing. this functionality is provided in `frontend/packages/integration-tests-cypress/views/yaml-editor.ts`